Option::<bool>::from_value_opt behavior

The following code prints `Some(true)`.

```
let value = Value::Int(1);
let out = Option::<bool>::from_value_opt(value).unwrap();
println!("{:?}", out);
```

The following code panics:
```
let value = Value::UInt(1);
let out = Option::<bool>::from_value_opt(value).unwrap();
println!("{:?}", out)
```

Is it expected that `Int(1)` can be converted to `Option<bool>` while `UInt(1)` can't?

BTW I'm on version 0.25. The newer versions fail to compile (a bunch of tokio errors):

```
    Updating crates.io index
   Compiling mysql_async v0.26.0
error[E0425]: cannot find function `spawn` in crate `tokio`
  --> /Users/wqfish/.cargo/registry/src/github.com-1ecc6299db9ec823/mysql_async-0.26.0/src/conn/pool/ttl_check_inerval.rs:59:24
   |
59 |                 tokio::spawn(idling_conn.conn.disconnect().then(move |_| {
   |                        ^^^^^ not found in `tokio`
   |
help: consider importing one of these items
   |
9  | use crate::conn::pool::mpsc::list::thread::spawn;
   |
9  | use std::thread::spawn;
   |
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -27,7 +27,7 @@ pin-project = "1.0.2"
 serde = "1"
 serde_json = "1"
 thiserror = "1.0.4"
-tokio = { version = "1.0", features = ["io-util", "fs", "net", "time"] }
+tokio = { version = "1.0", features = ["io-util", "fs", "net", "time", "rt"] }
 tokio-util = { version = "0.6.0", features = ["codec"] }
 tokio-native-tls = "0.3.0"
 twox-hash = "1"

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -33,6 +33,7 @@ jobs:
         condition: and(succeeded(), eq(variables['installRust.toolchain'], 'stable'))
         displayName: cargo fmt
       - bash: |
+          cargo +nightly build -Zfeatures=dev_dep
           SSL=false COMPRESS=false cargo test
           SSL=true COMPRESS=false cargo test
           SSL=false COMPRESS=true cargo test

